File: src/main/java/ast/ASTAssignExpr.java
package ast;

public class ASTAssignExpr extends ASTExpr {

    public ASTExpr lvalue;
    public ASTExpr rvalue;

    public ASTAssignExpr(ASTFileLocation loc, ASTExpr left, ASTExpr right){
        super(loc);
        lvalue = left;
        rvalue = right;
    }

    public void print(int i){
        printIndent(i);
        System.out.println("ASTAssignExpr:");
        lvalue.print(i+1);
        rvalue.print(i+1);
    }
}

File: src/main/kotlin/ast/ASTAssignExpr.kt
package ast

class ASTAssignExpr(loc: ASTFileLocation, val lvalue: ASTExpr, val rvalue: ASTExpr) : ASTExpr(loc) {

    override fun print(i: Int) {
        printIndent(i)
        println("ASTAssignExpr:")
        lvalue.print(i + 1)
        rvalue.print(i + 1)
    }
}